app/components/ChartSettings/index.js

```diff
@@ -52,19 +64,16 @@ class ChartSettings extends Component {
     const module = require(`./modules/custom/${config.settingsComponent}`).default;
     return React.createElement(module, {
       options: this.props.options,
-      defaultExpand: this._shouldDefaultExpand(),
+      defaultExpand: this._shouldDefaultExpand(name),
+      toggleCallback: (expanded) => this._toggleModuleCallback(expanded, name),
     });
   }
 
   render() {
     return (
       <div>
         <div>
-          {this._renderModule('XAxis')}
-          {this._renderModule('YAxis')}
-          {this._renderModule('Legend')}
-          {this._renderModule('Metadata')}
-          {this._renderModule('ColorPalette')}
+          {this.state.modules.map((name) => this._renderModule(name))}
```

Comment:
I'm opting to revert this change for now, but Guessing you are referring to something like this:

```const listItems = numbers.map((number) =>
  <li key={number.toString()}>
    {number}
  </li>
);```
https://guides.github.com/features/mastering-markdown/